pkg/rpc/adminservice/execution.go

```diff
@@ -142,6 +142,31 @@ func (m *AdminService) GetExecution(
 	return response, nil
 }
 
+func (m *AdminService) UpdateExecution(
+	ctx context.Context, request *admin.ExecutionUpdateRequest) (*admin.ExecutionUpdateResponse, error) {
+	defer m.interceptPanic(ctx, request)
+	requestedAt := time.Now()
```

Comment:
can we use this requestedAt as the state change occurred at value?

Reply:
Yes . works well. Also it helps in better unit testing the execution manager as this value is injected